### PR TITLE
382 add define-xml-path

### PR DESCRIPTION
**Pull request by mdeforestbrown**
Not 100% sure on the scope of this change, it seems we point to the define-xml from a few different places so there are quite a few changes here. Let me know if I'm on the right track and how best to add more tests for this.

### CODE REVIEW

File: core.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def cli():
     multiple=True,
     help="Absolute path to dataset file",
 )
+@click.option("-dxp", "--define-xml-path", default="test", help="Path to Define-XML")
```
Comment (nhaydel) [inline core.py:71]: I don't think we want a default here. I think we should add required=False

**Review by nhaydel — CHANGES_REQUESTED** [review]
I really like this. Only one small comment about the flag. Also there is one test failing, are you able to see that?

**Comment by nhaydel** [comment]
@mdeforestbrown just need to fix the lint errors. You can install the linter in the github repo by running `pre-commit install` in the root of the repo

File: scripts/run_validation.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ def validate_single_rule(cache, datasets, args: Validation_args, rule: dict = No
         ct_package=args.controlled_terminology_package,
         meddra_path=args.meddra,
         whodrug_path=args.whodrug,
+        define_xml_path=args.define_xml_path,
```
Comment (nhaydel) [inline scripts/run_validation.py:60]: Also, does the report get the correct define version if a define_xml_path is specified?

**Review by nhaydel — COMMENTED** [review]
I have a couple questions. Assuming those are handled, it should be good to go

File: cdisc_rules_engine/dataset_builders/dataset_builder_factory.py [hunk 3]
```diff
@@ -76,5 +76,6 @@ def get_service(
             kwargs.get("dataset_path"),
             kwargs.get("datasets"),
             kwargs.get("domain"),
+            kwargs.get("define_xml_path"),
```
[the remark this answers is not in the document]
Comment (mdeforestbrown) [reply cdisc_rules_engine/dataset_builders/dataset_builder_factory.py:79]: Oh yes my mistake, added.

File: scripts/run_validation.py (hunk 2 above)
Comment (mdeforestbrown) [reply scripts/run_validation.py:60]: Okay I have updated as discussed yesterday. 

If I move the define.xml to a separate folder and point to it with the path command it is now populating correctly for both the excel and the json report.

Screenshot 2023-07-21 at 10 32 27
Screenshot 2023-07-21 at 10 34 14

File: tests/QARegressionTests/test_core/test_test_command.py [hunk 4]
```diff
@@ -28,6 +28,8 @@ def test_test_command_with_all_options(self):
                 "3.4",
                 "-dv",
                 os.path.join("tests", "resources", "report_test_data", "define.xml"),
+                "-dxp",
```
Comment (nhaydel) [inline tests/QARegressionTests/test_core/test_test_command.py:31]: Does this test fail without this argument? I don't think we want it to

Comment (mdeforestbrown) [reply tests/QARegressionTests/test_core/test_test_command.py:31]: > Does this test fail without this argument?

No it doesn't fail without this argument. I'd add a separate test but assumed it was covered by the other tests in this file without the argument?

**Review by nhaydel — APPROVED** [review]

**Pull request merged**